frontend/simple/views/containers/Chatroom/Chatroom.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <main class="c-chatroom">
+    <chat-nav :title="title" :searchPlaceholder="searchPlaceholder">
+      <slot name="nav"></slot>
+    </chat-nav>
+    <chat-main :summary="summary" :details="details" />
+  </main>
+</template>
+<style lang="scss" scoped>
+@import "../../../assets/sass/theme/index";
+
+.c-chatroom {
+  padding: 0;
+
+  @include phone {
+    display: block;
+    min-height: auto;
+  }
+
+  @include phablet {
+    flex-direction: row;
+    align-items: stretch;
+    height: 100vh;
+  }
+}
+</style>
+<script>
+import { groupA, privateMessagesSortedByTime, users, messageConversations } from './fakeStore.js'
+import ChatMain from '../../components/Chatroom/ChatMain.vue'
+import ChatNav from '../../components/Chatroom/ChatNav.vue'
+
+export default {
+  name: 'Chatroom',
+  components: {
+    ChatMain,
+    ChatNav
+  },
+  props: {
+    title: String,
+    searchPlaceholder: String
+  },
+  data () {
+    return {
+      config: {
+        isPhone: null
+      },
+      ephemeral: {
+        isLoading: true
+      }
+    }
+  },
+  created () {
+    // TODO #492 create a global Vue Responsive just for media queries.
+    var mediaIsPhone = window.matchMedia('screen and (max-width: 639px)')
+    this.config.isPhone = mediaIsPhone.matches
+    mediaIsPhone.onchange = (e) => { this.config.isPhone = e.matches }
+  },
+  computed: {
+    currentConversation () {
+      // NOTE - This should be $store responsability but for now
+      // I've used the $route just for static mocked layout purposes for messages
+
+      let currentId
+
+      // TODO ALL OF THIS...
+
+      if (this.$route.name === 'Messages' && !this.config.isPhone) {
+        // Open by default the first conversation without unread messages
+        for (let i = 0, l = privateMessagesSortedByTime.length; i < l; i++) {
+          if (users[privateMessagesSortedByTime[i]].unreadCount === 0) {
+            currentId = privateMessagesSortedByTime[i]
+            break
+          }
+        }
+
+        this.$router.push({ path: `/messages/${users[currentId].name}` })
+        this.$route.params.currentConversation = { type: 'messages', id: currentId }
```

Comment:
- I do not believe this is legal. The [vue-router](https://router.vuejs.org/guide/essentials/dynamic-matching.html) documentation does not suggest that creating keys on this object is allowed, and rather the opposite suggests that the object is supposed to be immutable.
- Also, if you want to pass parameters via the URL, the correct way to do it is [shown here](https://router.vuejs.org/guide/essentials/navigation.html), e.g. either of the following:

```js
router.push({ path: 'register', query: { plan: 'private' }})

// or via params, but `userId` must match the route definition key, e.g. `path: '/name/:userId'`
router.push({ name: 'user', params: { userId: 123 }})
```

Reply:
> That's why I've left the comment several times around saying this is a workaround.

But it's not even a workaround, it's just invalid. You can't add keys to `$route.params` like this. At least not according to the vue-router documentation. You have to do something else.

> Maybe I could change it to sbp() so it's easier to delete in the future when we have the real logic/data implemented and don't mess with vue-router for now... Yes, I'll do that.

If you want to use `okTurtles.data` namespace for this that's something you could do, however, I recommend you pass them in via `query: ..` params, because that way the URL can be bookmarked, passed around, and back/forward buttons will work.

Reply:
Okay I'll use the query then as workaround :)